prow/spyglass/spyglass_test.go

```diff
@@ -434,6 +434,279 @@ func TestJobPath(t *testing.T) {
 	}
 }
 
+func TestRunPath(t *testing.T) {
```

Comment:
Maybe a cool test would be to use `ProwJobSpec`, call the GCS library to get the upload path, then use your parser and ensure you get back the parts of the spec you used to generate the path in the first place

Reply:
Good idea - done!